### PR TITLE
[4.1] RavenDB-11194 Prevent from putting a database on a node that is down.

### PR DESCRIPTION
**Pull request by karmeli87**

### CODE REVIEW

File: src/Raven.Server/ServerWide/ServerStore.cs [hunk 1]
```diff
@@ -1473,6 +1474,63 @@ private static bool DatabaseNeedsToRunIdleOperations(DocumentDatabase database)
...
         }
 
+        public void AssignNodesToDatabase(ClusterTopology clusterTopology, DatabaseRecord record)
+        {
+            var topology = record.Topology;
+
+            Debug.Assert(topology != null);
+
+            var clusterNodes = clusterTopology.Members.Keys
+                .Concat(clusterTopology.Promotables.Keys)
```
Comment (ayende) [inline src/Raven.Server/ServerWide/ServerStore.cs]: We shouldn't add it to promotables. They may take a while to come up and actually figure that one out.
Only members / watchers should be there.

File: src/Raven.Server/ServerWide/ServerStore.cs [hunk 2]
```diff
@@ -1473,6 +1474,63 @@ private static bool DatabaseNeedsToRunIdleOperations(DocumentDatabase database)
...
+                .Concat(clusterTopology.Watchers.Keys)
+                .ToList();
+
+            if (record.Encrypted)
+            {
+                clusterNodes.RemoveAll(n => AdminDatabasesHandler.NotUsingHttps(clusterTopology.GetUrlFromTag(n)));
+                if (clusterNodes.Count == 0)
```
Comment (ayende) [inline src/Raven.Server/ServerWide/ServerStore.cs]: Need to compare to the replication factor here, no?

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/ServerWide/ServerStore.cs (hunk 1 above)
Comment (karmeli87) [reply src/Raven.Server/ServerWide/ServerStore.cs]: done

File: src/Raven.Server/ServerWide/ServerStore.cs (hunk 2 above)
Comment (karmeli87) [reply src/Raven.Server/ServerWide/ServerStore.cs]: OK, done.

**Pull request merged**